generic-file-system/impl/src/main/java/org/pentaho/platform/genericfile/BaseGenericFileProvider.java

```diff
@@ -119,31 +117,57 @@ private int getEffectiveExpandedMaxDepth( @NonNull GetTreeOptions options ) {
 
   private boolean shouldProcessExpandedPath( @NonNull GetTreeOptions options ) {
     return options.getFilter() != GetTreeOptions.TreeFilter.FILES
-      && options.getExpandedPath() != null
-      && options.getMaxDepth() != null
-      && owns( options.getExpandedPath() );
+      && options.getExpandedPaths() != null
+      && options.getMaxDepth() != null;
   }
 
-  private void expandPathInTrees( @NonNull List<BaseGenericFileTree> baseTrees,
-                                  @NonNull GetTreeOptions options )
+  private List<GenericFilePath> expandPathsInTrees( @NonNull List<BaseGenericFileTree> baseTrees,
+                                                    @NonNull GetTreeOptions options )
+    throws OperationFailedException {
+
+    assert options.getExpandedPaths() != null;
+
+    List<GenericFilePath> effectiveExpandedPaths = null;
+
+    for ( GenericFilePath expandedPath : options.getExpandedPaths() ) {
+      if ( owns( expandedPath ) ) {
+        if ( expandOwnPathInTrees( baseTrees, options, expandedPath ) ) {
+
+          if ( effectiveExpandedPaths == null ) {
+            effectiveExpandedPaths = new ArrayList<>();
+          }
+
+          effectiveExpandedPaths.add( expandedPath );
+        }
+      }
```

Comment:
For sure a matter of own preference and coding style, but just commenting to see if you want to keep it this way or use the `&&` in a single if statement.

Reply:
Thanks, that was a distraction! Addressed in https://github.com/pentaho/pentaho-scheduler-plugin/pull/243/commits/06073e9c9849c698e9f0fbb830f60f802959440e.